### DISCUSSION TITLE
Ship a library with spdlog, link against an app that already use spdlog.

### DISCUSSION THREAD

**Discussion by simone-gaia** [gabime/spdlog#1937] [Q&A]
Hello, I have the following scenario: We are building a library that, amongst other things, provides a logger, which internally uses spdlog. We use the logger both to log internal stuff and to let our users log from their app. Our library is statically compiled, hence part of spdlog is statically compiled into it.

Now let's assume a user of our library already uses spdlog, or use another library that uses spdlog (eg. ros2). There can be a version mismatch between "our" spdlog and "their" spdlog, which can cause trouble (anything from SEGFAULT to weird behavior in the configuration of the logger).

I wonder if there is some best practice around this scenario or if what I'm doing is completely wrong.

Thanks.

**Comment by tt4g** [comment]
If there is no application binary compatibility between two spdlog, undefined behavior is expected to occur at runtime.
If you want to support applications that may use spdlog, it is recommended to provide both static library and shared libraries.

**Reply by tt4g** [reply]
Answers to similar questions found on Stack Overflow: https://stackoverflow.com/a/2216229
